petastorm/spark/spark_dataset_converter.py

```diff
@@ -231,40 +228,42 @@ def __init__(self,
                  data_url,
                  batch_size,
                  prefetch,
-                 preproc_fn,
-                 preproc_parallelism):
+                 num_epochs,
+                 workers_count,
+                 petastorm_reader_kwargs):
         """
         :param data_url: A string specifying the data URL.
         :param batch_size: batch size of the generated tf.data.dataset
         :param prefetch: prefetch for tf dataset
-        :param preproc_fn: preprocessing function
-        :param preproc_parallelism: parallelism for preprocessing function
         """
         from petastorm.tf_utils import make_petastorm_dataset
         import tensorflow as tf
 
         def support_prefetch_and_autotune():
             return LooseVersion(tf.__version__) >= LooseVersion('1.14')
 
-        self.reader = petastorm.make_batch_reader(data_url)
-        self.dataset = make_petastorm_dataset(self.reader) \
-            .flat_map(tf.data.Dataset.from_tensor_slices) \
+        if 'dataset_url' in petastorm_reader_kwargs:
+            raise ValueError('Cannot set dataset_url argument by yourself.')
```

Comment:
If we don't check it, petastorm will throw the error: `TypeError: make_batch_reader() got multiple values for argument 'dataset_url'`. This seems not necessary.

Reply:
@liangz1 Oh. Right. But this error message will make our user confusing. (User will say I only set one `dataset_url` arguments ?)